#!/usr/bin/python
PACKAGE = "teraranger"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thermal_image_flip_h", bool_t, 0, "Flip Thermal Image horizontally", False)
gen.add("thermal_image_flip_v", bool_t, 0, "Flip Thermal Image vertically", False)
gen.add("thermal_image_interpolate", bool_t, 0, "Interpolate Thermal Image", False)
gen.add("thermal_image_autoscale", bool_t, 1, "Autoscale Thermal Image", True)
gen.add("manual_min_scaling", double_t, 2, "Lower scaling bound", -20.0, 0.0, 1000.0)
gen.add("manual_max_scaling", double_t, 2, "Upper scaling bound", -20.0, 0.0, 1000.0)

map_enum = gen.enum([gen.const("Dave", int_t, 0,"Set colormap to Dave"),
            gen.const("Ice", int_t, 1,"Set colormap to Ice"),
            gen.const("Ironbow", int_t, 2,"Set colormap to Ironbow"),
            gen.const("High_contrast", int_t, 3,"Set colormap to High_contrast"),
            gen.const("Whot", int_t, 4,"Set colormap to Whot")],
            "Set the colormap for RGB image topic")

gen.add("Map", int_t, 3, "Set the sensor mode", 0, 0, 4, edit_method=map_enum)

exit(gen.generate(PACKAGE, "teraranger_evo_thermal", "EvoThermal"))
